storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/BaseResourceAwareStrategy.java

```diff
@@ -164,6 +168,7 @@ public SchedulingResult schedule(Cluster cluster, TopologyDetails td) {
 
         //order executors to be scheduled
         List<ExecutorDetails> orderedExecutors = execSorter.sortExecutors(unassignedExecutors);
+        isolateAckersToEnd(orderedExecutors);
```

Comment:
this seems to be a behavior change?  Why was this added?

Reply:
This is refactoring. This reordering was earlier being done in scheduleExecutorsOnNodes(). But it should be done when the executors are sorted. Then Round Robin strategy to can override scheduleExecutorsOnNodes() and still receive the executors sorted in right order (i.e. ackers at the end)